phoenix-core/src/main/java/org/apache/phoenix/schema/MetaDataClient.java

```diff
@@ -4362,18 +4365,40 @@ public MutationState alterIndex(AlterIndexStatement statement) throws SQLExcepti
                     // Set so that we get the table below with the potentially modified rowKeyOrderOptimizable flag set
                     indexRef.setTable(result.getTable());
                     if (newIndexState == PIndexState.BUILDING && isAsync) {
-                        try {
-                            tableUpsert = connection.prepareStatement(UPDATE_INDEX_REBUILD_ASYNC_STATE);
-                            tableUpsert.setString(1,
-                                    connection.getTenantId() == null ? null : connection.getTenantId().getString());
-                            tableUpsert.setString(2, schemaName);
-                            tableUpsert.setString(3, indexName);
-                            tableUpsert.setLong(4, result.getTable().getTimeStamp());
-                            tableUpsert.execute();
-                            connection.commit();
-                        } finally {
-                            if (tableUpsert != null) {
-                                tableUpsert.close();
+                        if (isRebuildAll) {
+                            List<Task.TaskRecord> tasks = Task.queryTaskTable(connection, schemaName, tableName, PTable.TaskType.INDEX_REBUILD,
```

Comment:
@gokceni Is tableName == data-table-name or index-table-name? If the former, how do we handle concurrent rebuilding of multiple indexes? I noticed that the check on line 4371 prevents multiple tasks being added to the Tasks table.

Reply:
It seems like it is data table name. This is the ALTER INDEX statement though. We are changing 1 index and rebuilding it. The task table check makes sure that we don't already have this in the task table (index can be built from the gus case, we don't want to kick one from gus and run Alter with rebuild too) The task table record will get deleted in 10 days (TTL).

Reply:
@gokceni My question was about multiple indexes on the same data-table. Since you say that the tableName is data-table, does this check prevent running ALTER-INDEX REBUILD ALL on two or more of indexes on the same data-table?

Reply:
queryTaskTable queries Task table with given parameters. The parameters include both index name and table name so, it checks if we have a task for building given index on the given table. If there is another task that builds another index on the same table, tasks.size will be equal to 0. I hope this answers your question

Reply:
Yes. Thank you @gokceni!